misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java

```diff
@@ -200,20 +197,55 @@ protected JsonElement transformSyntaxNode(Node node) {
             }
 
             nodeJson.add("typeData", symbolJson);
-            if (node.kind() == SyntaxKind.FUNCTION_BODY_BLOCK
-                    && (this.visibleEpsForEachBlock.size() > 0 || this.visibleEpsForModule.size() > 0)
-                    && nodeJson.get("typeData") != null) {
-                JsonArray eps = new JsonArray();
-                this.visibleEpsForEachBlock.forEach(eps::add);
-                this.visibleEpsForModule.forEach(eps::add);
-                nodeJson.add("VisibleEndpoints", eps);
-                this.visibleEpsForEachBlock = new ArrayList<>();
+
+            if ((node.kind() == SyntaxKind.BLOCK_STATEMENT || node.kind() == SyntaxKind.FUNCTION_BODY_BLOCK)
+                    && (this.visibleEpsForEachBlock.size() > 0 || this.visibleEpsForModule.size() > 0)) {
+
+                JsonArray blockEndpoints = new JsonArray();
+                // Add module level endpoints
+                this.visibleEpsForModule.forEach(blockEndpoints::add);
+
+                if (this.visibleEpsForEachBlock.size() > 0) {
+                    for (JsonObject endpoint : this.visibleEpsForEachBlock) {
```

Comment:
`for` block condition already checks if the visibleEpsForEachBlock size is greater than 0. We can avoid the `if` check.

Reply:
Sure